torchtune/utils/memory.py

```diff
@@ -15,14 +16,77 @@
 from torch.distributed.algorithms._checkpoint.checkpoint_wrapper import (
     apply_activation_checkpointing,
 )
+from torch.distributed.fsdp import FullyShardedDataParallel as FSDP
 from torch.distributed.fsdp.wrap import ModuleWrapPolicy
+from torchtune import modules
 from torchtune.utils.logging import get_logger
 
 _log: logging.Logger = get_logger()
 
 ACWrapPolicyType: Type = Union[Set[Type], Callable[[nn.Module, bool, int], bool]]
 
 
+def get_ac_policy(
+    memory_efficient_wrapping: bool, modules_to_wrap: Set[Type]
+) -> ACWrapPolicyType:
```

Comment:
I know some of this isn't introduced by this PR, but this is a really dense function which requires way more documentation, comments and explanation.

I have no idea what this means:
```
Union[Set[Type], Callable[[nn.Module, bool, int], bool]]
```